### PR TITLE
Select element on click

### LINKED ISSUE #11: [TODO] Ability to click to select
Right now we can only select using drag. If we click (not drag) on something, it should select it

### PR DESCRIPTION
**Pull request by giovannigiordano**
Fixes #11

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -228,41 +228,58 @@ class App extends React.Component {
...
-                element.measure.actualBoundingBoxAscent +
-                element.measure.actualBoundingBoxDescent;
-              // Center the text
-              element.x -= element.measure.width / 2;
-              element.y -= element.measure.actualBoundingBoxAscent;
-              element.width = element.measure.width;
-              element.height = height;
-            }
+            if (this.state.elementType === "selection") {
+              const selectedElement = elements.find(el => {
```
Comment (vjeux) [inline src/index.js]: Same comment as the other pull request, we should be consistent in how we name variables if possible. I used “element” everywhere else so far. Happy to see it changed to something else

File: src/index.js [hunk 2]
```diff
@@ -228,41 +228,58 @@ class App extends React.Component {
           onMouseDown={e => {
             const x = e.clientX - e.target.offsetLeft;
             const y = e.clientY - e.target.offsetTop;
-            const element = newElement(this.state.elementType, x, y);
```
Comment (vjeux) [inline src/index.js]: This is really interesting approach to put it onMouseDown. In my mind I was thinking that we would have to do onMouseUp and detect whether we had moved and trigger the click behavior. I can’t see anything wrong on top of my head where it wouldn’t work so let’s ship it :)

**Comment by vjeux** [comment]
Looks like it conflicts with the other pull request. Feel free to rebase and squash to master

**Comment by vjeux** [comment]
I added you as owner of the project. Since it’s at a young stage I believe that the best way to make it succeed is via super quick iteration.

From this PR I trust your judgement. Feel free to send any code change as pull request and merge it yourself to make progress :)

Can’t wait to see where it’s going to go!

File: src/index.js (hunk 1 above)
Comment (giovannigiordano) [reply src/index.js]: I usually avoid shadowed variable declarations,`element` is already declared in the upper scope

Comment (vjeux) [reply src/index.js]: Oh I see. It’s probably a sign that it’s time to refactor some of this huge function in some way :)

Maybe having a helper for finding selected elements would be useful.

Anyway, this doesn’t need to be done in this PR.

File: src/index.js [hunk 3]
```diff
@@ -330,11 +340,8 @@ class App extends React.Component {
               if (this.state.elementType === "selection") {
                 if (isDraggingElements) {
                   isDraggingElements = false;
-                } else {
-                  // Remove actual selection element
-                  setSelection(draggingElement);
-                }
-                elements.pop();
```
Comment (vjeux) [inline src/index.js]: How is the selection background removed? Can’t find any replacement.

To repro: mouse down, drag, mouse up. The blue rectangle should be gone

Comment (giovannigiordano) [reply src/index.js]: I am checking it, I had a conflict

**Comment by giovannigiordano** [comment]
Now it should be ok, I've fixed the selection.

**Pull request merged**